lms/static/scripts/frontend_apps/index.js

```diff
@@ -1,29 +1,30 @@
-// Polyfills.
 import 'focus-visible';
-
-// Setup app.
 import { createElement, render } from 'preact';
 
 import { readConfig, Config } from './config';
 import BasicLtiLaunchApp from './components/BasicLtiLaunchApp';
 import CanvasOAuth2RedirectErrorApp from './components/CanvasOAuth2RedirectErrorApp';
 import FilePickerApp from './components/FilePickerApp';
 import { ClientRpc } from './services/client-rpc';
+import { GradingService, Services } from './services';
 
 /** @typedef {import('./services/client-rpc').ClientConfig} ClientConfig */
 
-const rootEl = document.querySelector('#app');
-if (!rootEl) {
-  throw new Error('#app container for LMS frontend is missing');
-}
-
-const config = readConfig();
-
 import { registerIcons } from '@hypothesis/frontend-shared';
 import iconSet from './icons';
-
 registerIcons(iconSet);
 
+// Read configuration embedded into page by backend.
+const config = readConfig();
+
+// Create services.
+const services = new Map(
+  /** @type {[Function, any][]} */ ([
```

Comment:
This typing differs from the `ServiceMap` type (`Function` key instead of `Class`, I think?). Can you help me understand?

Reply:
I'm being a little lazy here. This works because classes are functions. A more specific type for "something you can create with `new`" is `new (...args: any): any`.